server/src/main/java/org/elasticsearch/action/support/broadcast/BroadcastResponse.java

```diff
@@ -127,4 +129,21 @@ public void writeTo(StreamOutput out) throws IOException {
             exp.writeTo(out);
         }
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
If we want to make BroadcastResponse implementing ToXContentFragment then we need to bring in the logic from `RestActions.buildBroadcastShardsHeader()` (but maybe not the exceptions grouping thing) here? And then change the calling methods in `server` so that they use this toXContent() instead of buildBroadcastShardsHeader?

Reply:
Yes, I see that we have build `RefreshResponse` [in this way](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRefreshAction.java#L65) in server side ? So we don't even need to implement `ToXContentFragment` here ?

Reply:
So you want to reuse RestActions.buildBroadcastShardsHeader() in RefreshResponse.toXContent() instead?

Reply:
Hello @tlrx, srry for not having responded in time because I just want to see a bit more to discuss.I mean we don't need to make `BroadcastResponse` or `RefreshResonse` implementing `ToXContentFragment` (then to implement an `toXContent()` method), do we ? All the work has done here https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRefreshAction.java#L64-L66 ?

Reply:
> srry for not having responded in time because I just want to see a bit more to discuss.

Don't be sorry at all! You're helping to build a better `RestHighLevelClient` and it's time consuming, it's normal :) 

> I mean we don't need to make BroadcastResponse or RefreshResonse implementing ToXContentFragment (then to implement an toXContent() method), do we ?

Well, the `RestHighLevelClient` does not render the response objects so I agree that we don't really need them to add the Refresh API. In the high level client we only need to parse the XContent corresponding to a `ResfreshResponse` object and that's the role of the `RefreshResponse.fromXContent()` method. 

But we need to test this fromXContent method and we have to be sure that a RefreshResponse object will always be generated the same way, in a way that can be parsed by the fromXContent method. We also try to be a bit lenient when we parse back the XContent so that it does not break when new fields are added or when fields are shuffled in the XContent. If you search for `testToAndFromXContent()` in the source code you'll see many of tests like this, where we rely on the toXcontent/fromXcontent methods for testing parsing/rendering of request/response objects. This is something we do since the begining of the RestHighLevelClient coding and I should we need to continue like this. We are also ok to clean up parsing or rendering logic in core if that helps the client and makes things cleaner.

By having the XContent building logic in the `RestRefreshAction` we have a strong link between the Response object and the Rest layer (and all stuff brought by it). I think that we should be able to render a RefreshResponse by itself, so I'd love to see it implementing toXContent (or toXContentFragment if it's easier) at some point.